projects/packages/search/src/initializers/class-initializer.php

```diff
@@ -7,14 +7,178 @@
 
 namespace Automattic\Jetpack\Search;
 
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
 use WP_Error;
-
 /**
  * Base class for the initializer pattern.
  */
-abstract class Initializer {
+class Initializer {
+	/**
+	 * Store the instance.
+	 *
+	 * @var Initializer
+	 */
+	protected static $instance;
+
+	/**
+	 * WPCOM blog ID
+	 *
+	 * @var int
+	 */
+	protected $blog_id;
+
+	/**
+	 * Initialize and get instance
+	 */
+	public static function instance() {
```

Comment:
Sorry, I can't help but ask - do we really need to use Singleton here? If we're going to be using Config to initialize Search, we're guaranteed to have the initialization code run once.

Reply:
Thanks for the input @zinigor. That's a valid point. changed in 9d9af29e09